artifact/v3/header/token.hpp

```diff
@@ -0,0 +1,112 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#ifndef MENDER_ARTIFACT_V3_HEADER_TOKEN_HPP
+#define MENDER_ARTIFACT_V3_HEADER_TOKEN_HPP
+
+
+#include <unordered_map>
+#include <string>
+#include <regex>
+
+#include <common/log.hpp>
+#include <common/expected.hpp>
+
+const std::regex artifact_script_regexp {
+	"scripts/Artifact(Install|Reboot|Rollback|RollbackReboot|Failure)_(Enter|Leave)_[0-9][0-9](_\\S+)?",
+	std::regex_constants::ECMAScript};
+
+namespace mender {
+namespace artifact {
+namespace v3 {
+namespace header {
+namespace token {
+
+using namespace std;
+
+namespace tar = mender::tar;
+
+namespace log = mender::common::log;
+namespace error = mender::common::error;
+namespace expected = mender::common::expected;
+
+enum class Type {
+	Uninitialized = 0,
+	EOFToken,
+	Unrecognized,
+	HeaderInfo,
+	ArtifactScripts,
+	ArtifactHeaderTypeInfo,
+	ArtifactHeaderMetaData,
+};
+
+unordered_map<const Type, string> type_map {
```

Comment:
Could be an array, no need for a hash map.

Reply:
If you have an enum that starts with `0`, you can have this mapping in an array and use the enum value as an index in the array.